### PR TITLE
Add consistency checks for mesh_nx mesh_ny relative to mesh_file in gen_mksurfdata_namelist.py

### PR DESCRIPTION
**Pull request by slevisconsulting**
### Description of changes
If mesh_file contains mesh_nx mesh_ny info, read it from the file.
Else abort and require user to enter it at the command line.
Confirm that the product of mesh_nx x mesh_ny (user-entered or from mesh_file) = elementCount (from mesh_file).

Address #1791 here, so as to name fsurdat files by their site/grid name rather than by their NX x NY.

Also address #1776 here, so as to stop hardwiring the mpirun command according to --machine in gen_mksurfdata_jobscript_single.py. Now get the mpirun command from env_mach_specific.xml.

### Specific notes

Contributors other than yourself, if any: @ekluzek 

CTSM Issues Fixed (include github issue #): #1790 #1791 #1776 

Are answers expected to change (and if so in what way)? No.

Any User Interface Changes (namelist or namelist defaults changes)? No.

Testing performed, if any:
Running gen_mksurfdata_namelist.py for cases with and without --model_mesh in the user-entered arguments and confirming expected behavior.

### CODE REVIEW

**Comment by slevisconsulting** [comment]
@ekluzek this PR could be ready for review and merging UNLESS you think I should fold #1776 here, too. I wouldn't mind doing that rather than opening a separate PR for that. Whatever you think makes more sense.

**Comment by ekluzek** [comment]
@slevisconsulting this likely means you haven't pulled all the tags in both of your clones. If you fetch everything in both clones they will likely be consistent. The tags it's pointing to are the ESCOMP tags, so fetch from ESCOMP, and you might need to use the "-t" option to make sure you pull all the tags.

**Comment by slevisconsulting** [comment]
> @slevisconsulting this likely means you haven't pulled all the tags in both of your clones. If you fetch everything in both clones they will likely be consistent. The tags it's pointing to are the ESCOMP tags, so fetch from ESCOMP, and you might need to use the "-t" option to make sure you pull all the tags.

```
Interesting. The -t option returned this:

> git fetch -t escomp
From https://github.com/ESCOMP/ctsm
 ! [rejected]            alpha-ctsm5.2.mksrf.04_ctsm5.1.dev090 -> alpha-ctsm5.2.mksrf.04_ctsm5.1.dev090  (would clobber existing tag)
 ! [rejected]            alpha-ctsm5.2.mksrf.05_ctsm5.1.dev090 -> alpha-ctsm5.2.mksrf.05_ctsm5.1.dev090  (would clobber existing tag)
```

If it's not concerning, I can leave as is and make the next tag (after we go over my mods) from izumi in case it makes a difference.

**Comment by ekluzek** [comment]
That sounds like your local tag in your clone is in disagreement with the tag on ESCOMP. I would trust the tag on ESCOMP and be suspicious of the tag on your local clone. You could look into it in more detail to see which is right, and if the ESCOMP ones are actually correct. But, even if the tags are wrong we are unlikely to do anything about it for these branch tags. But, you probably don't want to have a tag on your local clone that disagrees with ESCOMP.

File: tools/mksurfdata_esmf/gen_mksurfdata_namelist.py [hunk 1]
```diff
@@ -274,8 +279,32 @@ def main ():
...
+        element_count = mesh_file.dimensions['elementCount'].size
+        if 'origGridDims' in mesh_file.variables:
+            orig_grid_dims = mesh_file.variables['origGridDims']
+            force_model_mesh_nx = orig_grid_dims[0]
+            force_model_mesh_ny = orig_grid_dims[1]
+            mesh_file.close()
+            important_msg = 'Found data for force_model_mesh_nx and ' \
+                            'force_model_mesh_ny in the mesh file so ' \
+                            'IGNORING ANY CORRESPONDING USER-ENTERED VALUES.'
+            logger.info(important_msg)
```
Comment (slevisconsulting) [inline tools/mksurfdata_esmf/gen_mksurfdata_namelist.py:294]: @ekluzek request:
if orig_grid_dims disagrees with user-entered values, then abort with error message
if agrees, go on

File: tools/mksurfdata_esmf/gen_mksurfdata_namelist.py [hunk 2]
```diff
@@ -433,14 +467,24 @@ def main ():
...
-                    f'valid grid values are {valid_grids}'
-        sys.exit(error_msg)
+        if res in valid_grids:
+            error_msg = 'ERROR: You have requested a valid grid for which ' \
+        '../../ccs_config/component_grids_nuopc.xml does not include a mesh ' \
+        'file. For a regular regional or 1x1 grid, you may generate the ' \
+        'fsurdat file using the subset_data tool instead. Alternatively ' \
+        'and definitely for curvilinear grids, you may generate ' \
+        'a mesh file using the workflow currently (2022/7/6) described in ' \
+        'https://github.com/ESCOMP/CTSM/issues/1773#issuecomment-1163432584'
```
Comment (slevisconsulting) [inline tools/mksurfdata_esmf/gen_mksurfdata_namelist.py:482]: @ekluzek comment:
add TODO that this workflow should be included in the User's Guide.
add similar comment in the issue.

**Pull request merged**